test/integration/test-transaction.js

```diff
@@ -39,19 +41,20 @@ describe("transaction", function() {
   it("transaction commit", function(done) {
     shareConn.commit();
     shareConn.query("SET autocommit=0", () => {
-      assert.equal(shareConn.info.status, 0);
+      assert.equal(shareConn.info.status & ServerStatus.STATUS_IN_TRANS, 0);
+      assert.equal(shareConn.info.status & ServerStatus.STATUS_AUTOCOMMIT, 0);
       shareConn.beginTransaction(err => {
         if (err) done(err);
-        assert.equal(shareConn.info.status, 1);
+        assert.equal(shareConn.info.status & ServerStatus.STATUS_IN_TRANS, 1);
         shareConn.query("INSERT INTO testTransaction values ('test')", err => {
           if (err) done(err);
-          assert.equal(shareConn.info.status, 1);
+          assert.equal(shareConn.info.status & ServerStatus.STATUS_IN_TRANS, 1);
           shareConn.commit(err => {
             if (err) done(err);
-            assert.equal(shareConn.info.status, 0);
+            assert.equal(shareConn.info.status & ServerStatus.STATUS_IN_TRANS, 0);
             shareConn.query("SELECT count(*) as nb FROM testTransaction", (err, rows) => {
               if (err) done(err);
-              assert.equal(shareConn.info.status, 33);
+              assert.equal(shareConn.info.status & ServerStatus.STATUS_IN_TRANS, 1);
```

Comment:
Bitching a little bit - it should be not 1 here and in other places, but ServerStatus.STATUS_IN_TRANS. Or just that the result !=0